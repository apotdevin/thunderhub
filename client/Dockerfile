# stage: 1
FROM node:11-alpine as build

WORKDIR /usr/src/client

COPY package.json /usr/src/client
COPY yarn.lock /usr/src/client
RUN yarn install --production=true

COPY . /usr/src/client
RUN yarn build

# stage: 2 â€” the production environment
FROM nginx:alpine
COPY /config/nginx.conf /etc/nginx/conf.d/default.conf
COPY --from=build /usr/src/client/build /usr/share/nginx/html/
EXPOSE 80
CMD ["nginx", "-g", "daemon off;"]