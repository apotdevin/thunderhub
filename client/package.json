{
    "name": "app",
    "version": "0.1.9",
    "private": true,
    "dependencies": {
        "@apollo/react-hooks": "^3.1.3",
        "@types/crypto-js": "^3.1.43",
        "@types/jest": "25.1.4",
        "@types/lodash.debounce": "^4.0.6",
        "@types/lodash.merge": "^4.6.6",
        "@types/lodash.sortby": "^4.7.6",
        "@types/node": "13.9.3",
        "@types/numeral": "^0.0.26",
        "@types/qrcode.react": "^1.0.0",
        "@types/react": "16.9.25",
        "@types/react-copy-to-clipboard": "^4.3.0",
        "@types/react-dom": "16.9.5",
        "@types/react-modal": "^3.10.5",
        "@types/react-qr-reader": "^2.1.2",
        "@types/react-router-dom": "^5.1.2",
        "@types/react-tooltip": "^3.11.0",
        "@types/styled-components": "^5.0.1",
        "@types/styled-react-modal": "^1.2.0",
        "@types/styled-theming": "^2.2.2",
        "@types/victory": "^33.1.4",
        "@types/zxcvbn": "^4.4.0",
        "apollo-boost": "^0.4.4",
        "base64url": "^3.0.1",
        "crypto-js": "^4.0.0",
        "date-fns": "^2.11.0",
        "graphql": "^14.6.0",
        "lodash.debounce": "^4.0.8",
        "lodash.merge": "^4.6.2",
        "lodash.sortby": "^4.7.0",
        "node-sass": "^4.13.0",
        "numeral": "^2.0.6",
        "qrcode.react": "^1.0.0",
        "react": "^16.13.0",
        "react-copy-to-clipboard": "^5.0.2",
        "react-dom": "^16.13.0",
        "react-qr-reader": "^2.2.1",
        "react-router-dom": "^5.1.2",
        "react-scripts": "3.4.1",
        "react-spinners": "^0.8.1",
        "react-spring": "^8.0.27",
        "react-toastify": "^5.4.1",
        "react-tooltip": "^4.1.2",
        "snyk": "^1.303.0",
        "styled-components": "^5.0.1",
        "styled-react-modal": "^2.0.0",
        "styled-theming": "^2.2.0",
        "typescript": "^3.8.3",
        "victory": "^34.1.3",
        "zxcvbn": "^4.4.2"
    },
    "scripts": {
        "start": "react-scripts start",
        "build": "react-scripts build",
        "test": "react-scripts test",
        "eject": "react-scripts eject",
        "storybook": "start-storybook -p 9009 -s public",
        "build-storybook": "build-storybook -s public",
        "deploy": "yarn build && aws s3 --profile EBFullAccess sync build/ s3://thunderhub-client"
    },
    "eslintConfig": {
        "extends": "react-app"
    },
    "browserslist": {
        "production": [
            ">0.2%",
            "not dead",
            "not op_mini all"
        ],
        "development": [
            "last 1 chrome version",
            "last 1 firefox version",
            "last 1 safari version"
        ]
    },
    "devDependencies": {
        "@storybook/addon-actions": "^5.3.17",
        "@storybook/addon-info": "^5.3.17",
        "@storybook/addon-knobs": "^5.3.17",
        "@storybook/addon-links": "^5.3.17",
        "@storybook/addon-viewport": "^5.3.17",
        "@storybook/addons": "^5.3.17",
        "@storybook/preset-create-react-app": "^2.1.0",
        "@storybook/react": "^5.3.17",
        "awesome-typescript-loader": "^5.2.1",
        "husky": "^4.2.3",
        "prettier": "2.0.1",
        "pretty-quick": "^2.0.1",
        "react-docgen-typescript-loader": "^3.7.1"
    },
    "husky": {
        "hooks": {
            "pre-commit": "pretty-quick --staged"
        }
    }
}
