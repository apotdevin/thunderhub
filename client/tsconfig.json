{
    "compilerOptions": {
        "outDir": "build/dist",
        "module": "esnext",
        "target": "es2018",
        "lib": ["dom", "dom.iterable", "esnext"],
        "sourceMap": true,
        "rootDirs": ["src"],
        "baseUrl": "src",
        "moduleResolution": "node",
        "allowJs": true,
        "skipLibCheck": true,
        "esModuleInterop": true,
        "allowSyntheticDefaultImports": true,
        "strict": true,
        "forceConsistentCasingInFileNames": true,
        "resolveJsonModule": true,
        "isolatedModules": true,
        "noEmit": true,
        "jsx": "react",
        "noImplicitReturns": true,
        "noImplicitThis": true,
        "noImplicitAny": true,
        "strictNullChecks": true,
        "suppressImplicitAnyIndexErrors": true,
        // "noUnusedLocals": true,
        "declaration": true,
        "experimentalDecorators": true,
        "emitDecoratorMetadata": true
    },
    "include": ["src"],
    "exclude": ["node_modules", "build", "scripts", "**/*.stories.tsx"]
}
