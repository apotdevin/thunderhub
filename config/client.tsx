/* eslint @typescript-eslint/no-var-requires: 0 */
import { IncomingMessage, ServerResponse } from 'http';
import { useMemo } from 'react';
import {
  ApolloClient,
  InMemoryCache,
  NormalizedCacheObject,
} from '@apollo/client';
import getConfig from 'next/config';
import possibleTypes from 'src/graphql/fragmentTypes.json';

const { publicRuntimeConfig } = getConfig();
const { apiUrl: uri } = publicRuntimeConfig;

let apolloClient: ApolloClient<NormalizedCacheObject> | undefined;

export type ResolverContext = {
  req?: IncomingMessage;
  res?: ServerResponse;
};

function createIsomorphLink(context: ResolverContext = {}) {
  if (typeof window === 'undefined') {
    const { SchemaLink } = require('@apollo/client/link/schema');
    const { schema } = require('server/schema');
    const { getContext } = require('server/schema/context');
    return new SchemaLink({
      schema,
      context: getContext(context),
    });
  } else {
    const { HttpLink } = require('@apollo/client/link/http');
    return new HttpLink({
      uri,
      credentials: 'same-origin',
    });
  }
}

function createApolloClient(context?: ResolverContext) {
  return new ApolloClient({
    credentials: 'same-origin',
    ssrMode: typeof window === 'undefined',
    link: createIsomorphLink(context),
    cache: new InMemoryCache({
      ...possibleTypes,
    }),
    defaultOptions: {
      query: {
        fetchPolicy: 'cache-first',
      },
    },
  });
}

export function initializeApollo(
  initialState: NormalizedCacheObject | null = null,
  context?: ResolverContext
) {
  const _apolloClient = apolloClient ?? createApolloClient(context);

  if (initialState) {
    const existingCache = _apolloClient.extract();
    _apolloClient.cache.restore({ ...existingCache, ...initialState });
  }
  if (typeof window === 'undefined') return _apolloClient;
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState: NormalizedCacheObject | null) {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
