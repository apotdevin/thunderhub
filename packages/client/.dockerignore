node_modules
.git
.gitignore
build