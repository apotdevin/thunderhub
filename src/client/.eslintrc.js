module.exports = {
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: './tsconfig.json',
    sourceType: 'module',
    tsconfigRootDir: __dirname,
  },
  plugins: ['@typescript-eslint/eslint-plugin'],
  extends: [
    'next/core-web-vitals',
    'plugin:@typescript-eslint/recommended',
    'plugin:prettier/recommended',
    'prettier',
  ],
  root: true,
  env: {
    browser: true,
    node: true,
    jest: true,
  },
  ignorePatterns: ['.eslintrc.js'],
  rules: {
    '@typescript-eslint/interface-name-prefix': 'off',
    '@typescript-eslint/explicit-function-return-type': 'off',
    '@typescript-eslint/explicit-module-boundary-types': 'off',
    '@typescript-eslint/no-explicit-any': 'off',
    '@typescript-eslint/no-unused-vars': 'error',
    // '@next/next/no-html-link-for-pages': ['error', '/src/client/pages/'],
  },
};
