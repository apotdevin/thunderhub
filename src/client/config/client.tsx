/* eslint @typescript-eslint/no-var-requires: 0 */
import { IncomingMessage, ServerResponse } from 'http';
import { useMemo } from 'react';
import {
  ApolloClient,
  createHttpLink,
  from,
  InMemoryCache,
  NormalizedCacheObject,
} from '@apollo/client';
import getConfig from 'next/config';
import possibleTypes from '../src/graphql/fragmentTypes.json';
import { onError } from '@apollo/client/link/error';
import { setContext } from '@apollo/client/link/context';

const { publicRuntimeConfig } = getConfig();
const { apiUrl: uri } = publicRuntimeConfig;

let apolloClient: ApolloClient<NormalizedCacheObject> | undefined;

export type ResolverContext = {
  req?: IncomingMessage;
  res?: ServerResponse;
};

function createApolloClient(authToken: string) {
  const httpLink = createHttpLink({ uri, credentials: 'include' });

  const authLink = setContext((_, { headers }) => {
    return {
      headers: {
        ...headers,
        authorization: authToken ? `Bearer ${authToken}` : '',
      },
    };
  });

  const link = from([
    onError(({ graphQLErrors, networkError }) => {
      if (process.env.NODE_ENV === 'production') return;
      if (graphQLErrors)
        graphQLErrors.forEach(({ message, locations, path }) =>
          console.log(
            `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
          )
        );
      if (networkError) console.log(`[Network error]: ${networkError}`);
    }),
    authLink.concat(httpLink),
  ]);

  return new ApolloClient({
    credentials: 'same-origin',
    ssrMode: typeof window === 'undefined',
    link,
    cache: new InMemoryCache({
      ...possibleTypes,
    }),
    defaultOptions: {
      query: {
        fetchPolicy: 'cache-first',
      },
    },
  });
}

export function initializeApollo(
  authToken: string,
  initialState: NormalizedCacheObject | null = null
) {
  const _apolloClient = apolloClient ?? createApolloClient(authToken);

  if (initialState) {
    const existingCache = _apolloClient.extract();
    _apolloClient.cache.restore({ ...existingCache, ...initialState });
  }
  if (typeof window === 'undefined') return _apolloClient;
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(
  authToken: string,
  initialState: NormalizedCacheObject | null
) {
  const store = useMemo(
    () => initializeApollo(authToken, initialState),
    [authToken, initialState]
  );
  return store;
}
