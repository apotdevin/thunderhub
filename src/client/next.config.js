const ymlEnv = {
  YML_ENV_1: process.env.YML_ENV_1 || '',
  YML_ENV_2: process.env.YML_ENV_2 || '',
  YML_ENV_3: process.env.YML_ENV_3 || '',
  YML_ENV_4: process.env.YML_ENV_4 || '',
};

const ssoEnv = {
  cookiePath: process.env.COOKIE_PATH || '',
  lnServerUrl: process.env.SSO_SERVER_URL || '',
  lnCertPath: process.env.SSO_CERT_PATH || '',
  macaroonPath: process.env.SSO_MACAROON_PATH || '',
  dangerousNoSSOAuth:
    process.env.DANGEROUS_NO_SSO_AUTH === 'true' ? true : false,
};

const sslEnv = {
  publicUrl: process.env.PUBLIC_URL || '',
  sslPort: process.env.SSL_PORT || '',
  sslSave: process.env.SSL_SAVE || '',
};

const accountConfig = {
  accountConfigPath: process.env.ACCOUNT_CONFIG_PATH || '',
};

const urls = {
  mempoolUrl: process.env.MEMPOOL_URL || 'https://mempool.space',
};

module.exports = {
  reactStrictMode: true,
  distDir: '../../.next',
  poweredByHeader: false,
  basePath: process.env.BASE_PATH || '',
  serverRuntimeConfig: {
    nodeEnv: process.env.NODE_ENV || 'development',
    logLevel: process.env.LOG_LEVEL || 'info',
    torProxy: process.env.TOR_PROXY_SERVER || '',
    ...ssoEnv,
    ...accountConfig,
    ...sslEnv,
    ...ymlEnv,
  },
  publicRuntimeConfig: {
    ...urls,
    nodeEnv: process.env.NODE_ENV || 'development',
    apiUrl: `${process.env.BASE_PATH || ''}/graphql`,
    apiBaseUrl: `${process.env.API_BASE_URL || ''}/graphql`,
    basePath: process.env.BASE_PATH || '',
    npmVersion: process.env.npm_package_version || '0.0.0',
    defaultTheme: process.env.THEME || 'dark',
    defaultCurrency: process.env.CURRENCY || 'sat',
    fetchPrices: process.env.FETCH_PRICES === 'false' ? false : true,
    fetchFees: process.env.FETCH_FEES === 'false' ? false : true,
    disableLinks: process.env.DISABLE_LINKS === 'true' ? true : false,
    disableLnMarkets: process.env.DISABLE_LNMARKETS === 'true' ? true : false,
    noVersionCheck: process.env.NO_VERSION_CHECK === 'true' ? true : false,
    logoutUrl: process.env.LOGOUT_URL || '',
  },
};
