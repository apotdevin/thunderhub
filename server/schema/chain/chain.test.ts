import testServer from 'server/tests/testServer';
import { gql } from '@apollo/client';

jest.mock('ln-service');

describe('Chain Resolvers', () => {
  test('getChainBalance', async () => {
    const getChainBalance = gql`
      query {
        getChainBalance
      }
    `;
    const { query } = testServer();
    const res = await query({
      query: getChainBalance,
    });
    expect(res.errors).toBe(undefined);
    expect(res).toMatchSnapshot();
  });
  test('getPendingChainBalance', async () => {
    const getPendingChainBalance = gql`
      query {
        getPendingChainBalance
      }
    `;
    const { query } = testServer();
    const res = await query({
      query: getPendingChainBalance,
    });
    expect(res.errors).toBe(undefined);
    expect(res).toMatchSnapshot();
  });
});
