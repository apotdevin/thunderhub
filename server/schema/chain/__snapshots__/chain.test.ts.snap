// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chain Resolvers getChainBalance 1`] = `
Object {
  "data": Object {
    "getChainBalance": 1000000,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Chain Resolvers getPendingChainBalance 1`] = `
Object {
  "data": Object {
    "getPendingChainBalance": 1000000,
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
